contracts/LifCrowdsale.sol

```diff
@@ -131,17 +150,23 @@ contract LifCrowdsale is Ownable, Pausable {
     // calculate token amount to be created
     uint256 tokens = weiAmount.mul(rate);
 
-    // update state
-    weiRaised = weiRaised.add(weiAmount);
-    purchases[beneficiary] = weiAmount;
-    tokensSold = tokensSold.add(tokens);
+    // check if we are in the public presale block range
+    if (rate == publicPresaleRate) {
```

Comment:
You are right, so we can check if its in public presale or ICO on the fallback function and call buyTokens or buyTokensPresale, what do you think?